beep/generate_protocol.py

```diff
@@ -89,96 +108,88 @@ def generate_protocol_files_from_csv(csv_filename, output_directory=None):
     message = {"comment": "", "error": ""}
     if output_directory is None:
         output_directory = PROCEDURE_TEMPLATE_DIR
+
     for index, protocol_params in protocol_params_df.iterrows():
         template = protocol_params["template"]
+        protocol = None
         # Filename for the output
         filename_prefix = "_".join(
             [
                 protocol_params["project_name"],
                 "{:06d}".format(protocol_params["seq_num"]),
             ]
         )
-
-        # Switch for template invocation
-        if template == "EXP.000":
-            protocol = Procedure.from_exp(
-                **protocol_params[["cutoff_voltage", "charge_rate", "discharge_rate"]]
-            )
-            filename = "{}.000".format(filename_prefix)
-            filename = os.path.join(output_directory, "procedures", filename)
-        elif template == "diagnosticV2.000":
-            diag_params_df = pd.read_csv(
-                os.path.join(PROCEDURE_TEMPLATE_DIR, "PreDiag_parameters - DP.csv")
-            )
-            diagnostic_params = diag_params_df[
-                diag_params_df["diagnostic_parameter_set"]
-                == protocol_params["diagnostic_parameter_set"]
-            ].squeeze()
-
-            # TODO: should these be separated?
-            protocol = Procedure.from_regcyclev2(protocol_params)
-            protocol.add_procedure_diagcyclev2(
-                protocol_params["capacity_nominal"], diagnostic_params
-            )
-            filename = "{}.000".format(filename_prefix)
-            filename = os.path.join(output_directory, "procedures", filename)
-        # Each of these templates below has minor fixes to the avoid issues
-        # V3 fixes an error where the second portion of the charge gets skipped if the first charge rate is high
-        # V4 fixes the safety limits for the lower voltage
-        # V5 fixes the exit step for the safety condition during the final diagnostic and the storage cycle
-        elif template in ["diagnosticV3.000", "diagnosticV4.000", "diagnosticV5.000"]:
-            diag_params_df = pd.read_csv(
-                os.path.join(PROCEDURE_TEMPLATE_DIR, "PreDiag_parameters - DP.csv")
-            )
-            diagnostic_params = diag_params_df[
-                diag_params_df["diagnostic_parameter_set"]
-                == protocol_params["diagnostic_parameter_set"]
-            ].squeeze()
+        if ".000" in template:
             template_fullpath = os.path.join(PROCEDURE_TEMPLATE_DIR, template)
-            if protocol_params["project_name"] == "RapidC":
+            template_length = template_detection(template_fullpath)
+            if "diagnostic_parameter_set" in protocol_params:
```

Comment:
Can we put in some comments next to each of these `if`'s and `elif`'s to describe what we are matching.

Reply:
Good idea! Added those now